src/components/container/EhrNewsFeedEventDetail/EhrNewsFeedEventDetail.tsx

```diff
@@ -0,0 +1,156 @@
+import { EhrNewsFeedClaimProcedureModel, EhrNewsFeedClaimServiceModel, EhrNewsFeedEventModel, EhrNewsFeedEventType, EhrNewsFeedFeed, EhrNewsFeedLabReportModel, EhrNewsFeedProcedureModel } from '../../../helpers/news-feed/types';
+import "./EhrNewsFeedEventDetail.css"
+import language from '../../../helpers/language';
+
+export interface EhrNewsFeedEventDetailProps {
+    feed: EhrNewsFeedFeed
+    pageId?: string
+    pageDate?: string
+    previewState?: EhrNewsFeedEventType
+    onViewLabObservationTermInfo(labObservationID: string): void
+}
+
+export default function (props: EhrNewsFeedEventDetailProps) {
+    let [loading, setLoading] = useState<boolean>(false);
+    let [event, setEvent] = useState<EhrNewsFeedEventModel | null>(null);
+
+    function load() {
+        if (props.previewState) {
+            let handler = eventTypeDefinitions[props.previewState!];
+            setEvent(handler.getPreviewEvent());
+            return;
+        }
+
+        setLoading(true);
+        getNewsFeedPage(props.feed, props.pageId, props.pageDate).then((result) => {
+            setEvent(result.Events[0]);
+            setLoading(false);
+        });
+    }
+
+    useEffect(() => {
+        load();
+    }, []);
+
+    return <div className="mdhui-ehr-news-feed-event-detail">
+        {loading &&
+            <LoadingIndicator />
+        }
+        {event &&
+            <>
+                <NewsFeedDetailTitle event={event} />
+                {event.Type == "ProcedureGroup" && <ProcedureGroupDetail event={event} />}
+                {event.Type == "ClaimProcedureGroup" && <ClaimProcedureGroupDetail event={event} />}
+                {event.Type == "LabReport" && <LabReportDetail event={event} onViewLabObservationTermInfo={props.onViewLabObservationTermInfo} />}
+                {event.Type == "ClaimServiceGroup" && <ClaimServiceGroupDetail event={event} />}
+            </>
+        }
+    </div>
+}
+
+function NewsFeedDetailTitle(props: { event: EhrNewsFeedEventModel }) {
+    let handler = eventTypeDefinitions[props.event.Type];
+
+    let title = handler.getDetailTitle ? handler.getDetailTitle(props.event) : undefined;
+    if (!title) { return null; }
+
+    let date = parseISO(props.event.Date);
+    let subtitle = `${format(parseISO(props.event.Date), "EEEE, MMMM do, y h:mm a")} • ${props.event.Patient.RecordAuthority}`
+    if (format(date, "h:mm a") == "12:00 AM") {
+        subtitle = props.event.Patient.RecordAuthority;
+    }
+
+    return <Action
+        indicator={<></>}
+        title={title}
+        subtitle={subtitle}
+        icon={<img src={handler.icon} width={24} />} />;
+}
+
+function ProcedureGroupDetail(props: { event: EhrNewsFeedEventModel }) {
+    let procedures = props.event.Event as EhrNewsFeedProcedureModel[];
+    return <>
+        {procedures.map((procedure, index) =>
+            <Card key={procedure.ID} style={{ marginTop: "0" }}>
+                <Title defaultMargin order={5}>{procedure.Procedure}</Title>
```

Comment:
The use of the order property here is a  bit obscure and could benefit from some contextual documentation. Also think the simple enum pattern does not work as well with numeric values as it does with strings because they aren't self documenting. Maybe would consider a true enum for that order property.

Reply:
This Title component prop names were borrowed from some other UI frameworks so it seems conventional, e.g. https://mantine.dev/core/title/

Not sure what you mean about not being self-documenting though, it will show red squigglies etc. if you try to use one not in the list...

Reply:
Ah, when I looked at the class I saw this:
```
    if (props.image && props.autosizeImage) {
        imageWidth = 24;
        switch (props.order) {
            case 1:
                imageWidth = 40;
                break;
            case 2:
                imageWidth = 30;
                break;
            case 3:
                imageWidth = 24;
                break;
            case 4:
                imageWidth = 20;
                break;
            case 5:
                imageWidth = 16;
                break;
            case 6:
                imageWidth = 12;
                break;
        }
        if (props.imageAlignment == "top") {
            imageWidth = imageWidth * 2;
        }
    }
```
And did not realize order meant title level. Thought it was just some weird formatting convention. 

By not self documenting, I mean there is no text with implied usage like you have for a string constant.